### ISSUE TITLE
Do you have an encoder to go from a decodedCall to a `Transaction`?

### ISSUE THREAD

**Issue by Tbaut** [polkadot-api/polkadot-api#740]
Hey hey,

For Multix where I'm digging into decoded Transactions, I'd like to go from a decoded Transaction (e.g `Transaction<any, any, any, any>['decodedCall]` to a `Transaction<any, any, any, any>`.

For instance, I need to calculate the hash and the encoded call data from this deeply nested `create_pure`.

![image](https://github.com/user-attachments/assets/757dcbf2-fa51-46f5-82ed-6bd7d18ce0dc)

In other words, I'd like to be able to do:
```ts
const call= {
    "type": "Proxy",
    "value": {
        "type": "create_pure",
        "value": {
            "proxy_type": {
                "type": "Any"
            },
            "delay": 0,
            "index": 0
        }
    }
}

const tx: Transaction<any, any, any, ant> = client.txFromDecodedCall(call, compatibilityToken)

```

**Comment by carlosala** [comment]
It is actually feasible but a bit difficult at this point. It could be a good addition though (making it easier, I mean).

**Comment by Tbaut** [comment]
For anyone needing a workaround, Josep showed me the following somewhat hacky way, but it'll most probably always work. Create a batch call, then remove the 3 bytes from the batch encoded call data.
```ts
import { CompatibilityToken, Transaction } from 'polkadot-api'
import { ApiType } from '../contexts/ApiContext'

export const getEncodedCallFromDecodedTx = (
  api: ApiType,
  decodedTx: Transaction<any, any, any, any>['decodedCall'],
  compatibilityToken: CompatibilityToken
) => {
  const batch = api.tx.Utility.batch({
    calls: [decodedTx]
  }).getEncodedData(compatibilityToken)

  // we slice the 3 bytes from utility batch
  return batch.asHex().replace('0x', '').slice(6)
}
```
